js/theme-switcher.js

```diff
@@ -0,0 +1,24 @@
+// Replacement for jQuery $(selector) and $.onready()
+const _ = s => document.querySelector(s);
+const ready = f => document.readyState !== 'loading' ? f() : document.addEventListener('DOMContentLoaded', f)
+
+const darkMode = window.matchMedia && window.matchMedia("(prefers-color-scheme: dark)")
+
+if (darkMode || (window.localStorage && localStorage.getItem("theme") === "night")) {
+    _("body").classList.add("night");
+}
```

Comment:
FWIW; I'm looking at some additional changes, and in a follow up may move this code inline, so that we perform the switch even earlier (preventing the "light flash" when we load).